server/routes/index.ts

```diff
@@ -883,6 +884,42 @@ export function defineRoutes(router: IRouter) {
   );
 }
 
+const wrapRouteWithDataSource = async (
+  dataSourceEnabled: boolean,
+  context: RequestHandlerContext,
+  request: OpenSearchDashboardsRequest<unknown, unknown, any>,
+  response: OpenSearchDashboardsResponseFactory,
+  endpoint: string
+) => {
+  if (!dataSourceEnabled || !request.body?.dataSourceId) {
+    const client = context.security_plugin.esClient.asScoped(request);
+    let esResponse;
+    try {
+      esResponse = await client.callAsCurrentUser(endpoint);
+      return response.ok({
+        body: {
+          message: esResponse.message,
+        },
+      });
+    } catch (error) {
+      return errorResponse(response, error);
+    }
+  } else {
+    const client = context.dataSource.opensearch.legacy.getClient(request.body?.dataSourceId);
+    let esResponse;
+    try {
+      esResponse = await client.callAPI(endpoint, {});
```

Comment:
Can this be renamed as `callAsCurrentUser` so that it matches the block above? Ideally, the only thing that is different is the http client between the 2 blocks.

Reply:
This is the equivalent to callAsCurrentUser from the legacy client if I am understanding correctly. Used this to stay consistent with the other parts of the plugin where we are using legacy clients. https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/docs/multi-datasource/client_management_design.md#421-legacy-client